azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -236,7 +237,7 @@ def get_vmagent_update_status(self):
                 else:
                     status = VMAgentUpdateStatuses.Error
                     code = 1
-                return VMAgentUpdateStatus(expected_version=str(self._updater.version), status=status, code=code, message=self._last_attempted_update_error_msg)
+                return VMAgentUpdateStatus(expected_version=str(CURRENT_VERSION), status=status, code=code, message=self._last_attempted_update_error_msg)
```

Comment:
I'm a little concerned about the straight conversion of the FlexibleVersion to str (should probably be ustr, btw).

*Should* not be an issue for Agent versions, but in general, Flexible Version can have a pretty arbitrary format that CRP may not be able to parse correctly.

I would do an explicit conversion of the W.X.Y.Z components of the flexible version

Reply:
Never mind, the concern was about DistroVersion. In fact, previously I recommended (see distroversion.py)
`
 NOTE: For entities with simpler version schemes (e.g. extensions and the Agent), use FlexibleVersion.`